24 q1

dariok opened the pull request.

dariok committed:
handle tei:foreign

dariok committed:
function.xqm needs to import pquery

dariok committed:
handle foreign/@style

dariok committed:
Templating function: add attribute to element

get the value by evaluating an XQuery expression:
data-template-attribute for the attrib’s name
data-template-expression for the code to evaluate

dariok committed:
use standard $ed for project ID

finalize and apply nomenclature for function parameters
adjust to style book

Cf. #499 #441

dariok committed:
file upload: show info on parser error

dariok committed:
display sub-structures of current project

dariok committed:
update README and description for v3.0

dariok committed:
enable XInclude processing by default

dariok committed:
Drop METS support

Cf. #418

dariok committed:
Entities: only look for list* in text

for now, we ignore standOff as this may return many result per collection
for the future, device a way to have magic file names (see #99 #183)

dariok committed:
use an XSLT to display entity data

To do: extend XSLT to return useful results
Based on ebda1d1 which somehow got lost

Closes #183 #110 #99

dariok committed:
Addins: main module must return one or more elements

dariok committed:
remove copy of github Wiki

a more in-depth documentation is planned
Cf. # 158

dariok committed:
fix a bug with paginated search results

- make sure the right value is submitted for the start index
- JS: decodeURI() does not decode %3A, while decodeURIComponent() does

dariok committed:
initial query from form must set start

dariok committed:
Error when a file from wdbmeta is not found

also, defer error handling from wdb:populateModel to caller

Cf. #304 #193

dariok committed:
transform files which point to peers

dariok committed:
For now, we do not index tei:w

For most intents and purposes, we need the structures only

dariok committed:
improved error handling in app.xqm and function.xqm

dariok committed:
support for HTTP HEAD requests

dariok committed:
evaluate If-Modified-Since

- create common functions
- use in rest-files
- update app.xqm to use this

Cf #549 #313

dariok committed:
new function: file’s collection and filename as map

Cf. #549 #478

WunschK commented on README.md:
```diff
@@ -102,6 +94,7 @@ See the Wiki for details!
   * Repertotium frühneuzeitlicher Rechtsquellen
```
Repertorium

WunschK reviewed (commented):
looks good to me. 

why delete the documentation, though?

dariok committed:
extend modification headers

- functions for path and id
- evaluate If-Modified-Since for nav.html

dariok committed:
dump error map from function.xqm

under certain circumstances, we need to do this explicitly

dariok committed:
some simplifications

dariok committed:
Add more info to licensing information

Currently, German only. This needs i18n

dariok committed:
get path from ID: use entry in wdbmeta onliy

dariok committed:
Explicitly copy index definitions

Closes #406

dariok committed:
getEdPath: use self for eficiency

should be up for replacement in a separate module, though
cf. wdb-files.xqm

dariok committed:
getFullPath must return the right path

dariok committed:
Simplify file path lookup

Cf. #549 #474 #408

dariok committed:
Improve user-facing error messages when ID not found

In order to do this, adjust try/error/catch in populateModel() and start()

Cf. #343 #336 #304 #193
Closes #502

dariok committed:
user correct parameter name in doc

dariok committed:
return 404 when no file with ID

cf. #343 #304 #193
closes #336

dariok committed:
add info about keys in returned map in getFullPath()
